components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -341,6 +341,92 @@ List<IdentityProvider> getIdPsSearch(int tenantId, List<ExpressionNode> expressi
+        }
+    }
+
+    /**
+     * Get the result set.
+     *
+     * @param dbConnection       database connection.
+     * @param sortedOrder        Sort order.
+     * @param tenantId           tenant Id of the identity provider.
+     * @param offset             offset value.
+     * @param limit              limit per page.
+     * @param filterQueryBuilder filter query buider object.
+     * @param requiredAttributes Required attributes which needs to be return.
+     * @return result set of the query.
+     * @throws SQLException                              Database Exception.
+     * @throws IdentityProviderManagementServerException Error when getting list of Identity Providers.
+     */
+    private ResultSet getTrustedJwtIssuerQueryResultSet(Connection dbConnection, String sortedOrder, int tenantId,
+                                                        int offset, int limit, FilterQueryBuilder filterQueryBuilder,
+                                                        List<String> requiredAttributes)
+            throws SQLException, IdentityProviderManagementServerException, IdentityProviderManagementClientException {
+
+        String sqlQuery;
+        String sqlTail;
+        PreparedStatement prepStmt;
+        Map<Integer, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
+        int filterAttributeValueSize = filterAttributeValue.entrySet().size();
+        String databaseProductName = dbConnection.getMetaData().getDatabaseProductName();
+        if (databaseProductName.contains("Microsoft")) {
+            sqlQuery = IdPManagementConstants.SQLQueries.GET_JWT_ISSUER_TENANT_MSSQL;
+            sqlQuery = appendRequiredAttributes(sqlQuery, requiredAttributes);
+            sqlTail = String.format(IdPManagementConstants.SQLQueries.GET_JWT_ISSUER_TENANT_MSSQL_TAIL, sortedOrder);
+            // Add the type filter to the query.
+            String jwtIssuerFilterSql = IdPManagementConstants.SQLQueries.JWT_ISSUER_FILTER_SQL;
+            filterQueryBuilder.setFilterQuery(filterQueryBuilder.getFilterQuery() + jwtIssuerFilterSql);
+            sqlQuery = sqlQuery + IdPManagementConstants.SQLQueries.FROM_JWT_ISSUER_WHERE
+                    + filterQueryBuilder.getFilterQuery() + sqlTail;
+            prepStmt = dbConnection.prepareStatement(sqlQuery);
+            for (Map.Entry<Integer, String> prepareStatement : filterAttributeValue.entrySet()) {
+                prepStmt.setString(prepareStatement.getKey(), prepareStatement.getValue());
+            }
+            prepStmt.setInt(filterAttributeValueSize + 1, tenantId);
+            prepStmt.setInt(filterAttributeValueSize + 2, offset);
+            prepStmt.setInt(filterAttributeValueSize + 3, limit);
+        } else {
+            String message = "Error while loading Identity Provider from DB: Database driver could not be identified" +
+                    " or not supported.";
+            log.error(message);
```

Comment:
Doesn't this print the error log twice? Because of logging the error and throwing it.